core/src/main/java/io/questdb/cairo/TableReaderRecord.java

```diff
@@ -163,6 +160,26 @@ public long getLong(int col) {
         return reader.getColumn(absoluteColumnIndex).getLong(offset);
     }
 
+    @Override
+    public long getLong128Hi(int col) {
```

Comment:
This API forces us to implement suboptimal implementation. We have to do this column memory lookup twice. Have you considered implementing the `Long128` value type? Similar to existing `Long256`.

Reply:
yes. here are the results: 

We have 3 different access methods:
* Double address calculation method - the current impl
* Location method - basically exposing the address in a 3rd method - as a user first get the address and then pass it to `getHi()/getLo()` methods so the methods don't need to calculate it. 
* Wrapper method - long128 wrapper class

We have 4 sets of tests:
* SQL with parallel filters
* Single threaded SQL filter
* Direct access to Memory - calculating sum
* Access via Reader - calculating sum

The test #1 was executed via web console. All other tests were driven by JMH.
Tests #1 and #2 use 4B of rows and I executed them on a beefy EC2 machine.
Tests #3 and #4 use just 10M rows and I executed them on my local M1 Macbook.

Results:
Test #1: All 3 methods take 1s to scan all rows. There was no measurable differences between the methods! Apparently, the parallel filter hides it. Hypothesis: Something else (thread coordination??) dominates the cost and this dwarfs the cost of filter execution.
Test #2: Location: 10.855s, Double: 11.409s, Wrapper: 15.136s.
Test #3: Location:  14.929ms, Wrapper: 16.967 (the double method was not tested because there is no complex address calculation in this test)
Test #4: Location: 33.027ms, Double: 32.897ms, Wrapper: 40.358ms.
Summary:
The wrapper method is consistently the slowest method. Sometimes by a high margin. The only exception is the parallel SQL filters test, where all methods perform equally for some reason.